### PR TITLE
Vs code compilation experiments

### PR DESCRIPTION
**Pull request by aJonathanSchneider**
trying to resolve errors thrown in VS Code, packaging for npm
experiments:
- following package structure of @angular packages: index.d.ts is slimmed down, code moved to src\interfaces.ts
- 'Promise' wasn't found as a class, included rxjs-Promises
- 'String'-usage changed to 'string'
- imported interfaces where errors were thrown
- 'import {default}' threw an error, Classes directly imported

### CODE REVIEW

**Comment by tpluscode** [comment]
Oh wow, the changes are much more extensive than I had hoped for :)

I will leave you some code comments but I also have two more general questions. Please bear with me, because I'm far from expert on TS/ES6

1. Do you really find it necessary to have `tsconfig.json` and actually to do `tsc` compilation? Personally I only even compiled when bundling.
1. I'm not convinced about adding `rxjs` to have `Promise`. Again, at compilation it seems safe to ignore missing class. At runtime I don't want to enforce any specific implementation there of. I'd rather let the consumer decide to use native (where available) or choose one of the polyfills. And so, maybe simply installing global typings for `es6-promise` could to the trick?

File: typings.json [hunk 1]
```diff
@@ -3,6 +3,7 @@
     "core-decorators": "github:retyped/core-decorators-tsd-ambient/core-decorators.d.ts",
     "core-js": "registry:dt/core-js#0.0.0+20160317120654",
     "fetch.js": "github:ryan-codingintrigue/typescript-fetch/fetch.js.d.ts",
+    "node": "registry:dt/node#6.0.0+20160909174046",
```
Comment (tpluscode) [inline typings.json]: Why was this added?

File: package.json [hunk 2]
```diff
@@ -30,29 +35,69 @@
...
+    "core-js": "^2.4.1",
     "jsonld": "^0.4.6",
     "li": "^1.0.1",
+    "linkeddata-vocabs": "^1.0.0",
     "lodash": "^4.9.0",
     "rdf-ext": "^0.3.0",
     "rdf-formats-common": "^0.3.0",
-    "rdf-serializer-jsonld": "^0.3.0"
+    "rdf-serializer-jsonld": "^0.3.0",
+    "rxjs": "^5.0.0-rc.1",
```
Comment (tpluscode) [inline package.json]: Hm, and actually, `Promise` is included in `core-js` typings. Is that not enough for tsc?

File: package.json [hunk 3]
```diff
@@ -6,9 +6,14 @@
   "directories": {
     "test": "tests"
   },
+  "typescript": {
+    "definition": "index.d.ts"
+  },
+  "_where": "D:\\dev\\ws\\bizSys\\bizsys-prototype",
```
Comment (aJonathanSchneider) [inline package.json]: it seems that the "typescript" -part is needed, the other one can be removed

File: package.json [hunk 4]
```diff
@@ -6,9 +6,14 @@
...
   "scripts": {
-    "test": "./node_modules/karma/bin/karma start --single-run",
-    "build": "jspm bundle-sfx src/heracles - jsonld - li - lodash - core-js - core-decorators - rdf-ext - rdf-formats-common - rdf-serializer-jsonld dist/heracles.js --global-name heracles --format cjs --skip-source-maps"
+    "test": "karma start --single-run",
+    "build": "jspm bundle-sfx src/heracles - jsonld - li - lodash - core-js - core-decorators - rdf-ext - rdf-formats-common - rdf-serializer-jsonld dist/heracles.js --global-name heracles --format cjs --skip-source-maps",
+    "buildbig": "jspm bundle-sfx src/heracles + jsonld + li + lodash + core-js + core-decorators + rdf-ext + rdf-formats-common + rdf-serializer-jsonld dist/heracles.js --global-name heracles --format cjs --skip-source-maps"
```
Comment (aJonathanSchneider) [inline package.json]: yes I tried to bundle everything

File: typings.json (hunk 1 above)
Comment (aJonathanSchneider) [inline typings.json]: that was just an experiment to make this typings.json look similar to another project of mine

File: package.json (hunk 4 above)
Comment (tpluscode) [inline package.json]: Pretty scary right? :) I've just yesterday added a diet issue #9 to reduce the total size of heracles+dependencies

File: package.json (hunk 3 above)
Comment (tpluscode) [inline package.json]: Yup, I left that in

File: package.json (hunk 4 above)
Comment (aJonathanSchneider) [inline package.json]: yes it was about 40k lines of code, so I dropped that idea :P By the way, when developing, did you rather have the backend or the frontend in mind?

Comment (tpluscode) [inline package.json]: I do have the front end in mind. Optimization is definitely in order. Do you have any suggestions?
